src/Emulator.ts

```diff
@@ -118,14 +134,24 @@ implements IGetGenesisInfos, IGetProtocolParameters, IResolveUTxOs, ISubmitTx
                     this.pushUtxo(new UTxO({
                         resolved: tx.body.outputs[i],
                         utxoRef: new TxOutRef({
-                            id: new Hash32('a'.repeat(64)),
+                            id: tx.hash.toString(), // to be fixed
                             index: i
                         })
                     }))
                 }
+                
             }
+            // to do - update this.slot and this.time
+            // to do - Move time forward to the next block number
+            // to do - Update the block number
+
+            // Update slot and time
+            
+            console.log(`Advanced to block number ${height} (slot ${this.slot}). Time: ${new Date(this.time).toISOString()}`);
```

Comment:
```suggestion
            console.log(`Advanced to block number ${height} (slot ${this.slot}). Time: ${new Date(this.time).toISOString()}`);
```
This is wrong, you moved by `height` block number, not to `height`. You can either recompute the corresponding block number from the time or slot, or just keep track of it, as they do in Lucid, in a separate variable